src/server/worker/datum/datum.go

```diff
@@ -447,12 +420,48 @@ func (d *Datum) handleSymlinks(mf client.ModifyFile, storageRoot string) error {
 				input = i
 			}
 		}
+		// Upload the local files if they are not using the empty files feature.
+		if !input.EmptyFiles {
+			return d.uploadSymlink(mf, dstPath, file, fi)
+		}
 		srcFile := proto.Clone(input.FileInfo.File).(*pfs.File)
 		srcFile.Path = path.Join(pathSplit[1:]...)
 		return mf.CopyFile(dstPath, srcFile, client.WithDatumCopyFile(d.ID))
 	})
 }
 
+func (d *Datum) uploadSymlink(mf client.ModifyFile, dstPath, file string, fi os.FileInfo) error {
+	cb := func(dstPath, file string) (retErr error) {
+		f, err := os.Open(file)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err := f.Close(); retErr == nil {
+				retErr = err
+			}
+		}()
+		return mf.PutFile(dstPath, f, client.WithDatumPutFile(d.ID))
+	}
+	if fi.IsDir() {
+		dir := file
+		return filepath.Walk(dir, func(file string, fi os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if fi.IsDir() {
+				return nil
```

Comment:
@brycemcanally why are we returning nil here? my naive expectation would be that we search for files recursively

Reply:
also I realize this logic was here before, but figured it's a good time to ask

Reply:
We skip directories here because we cannot upload directories through PFS, they are created implicitly based on files. We upload the files, and if they have a directory prefix, the directories will be created. Empty directories will be skipped because PFS does not support empty directories.